### PR TITLE
Run tests on PHP 8.3 and update test suite

### PR DESCRIPTION
**Pull request by yadaiio**
Builds on top of #30, #35, #36 and #37.

References: https://github.com/reactphp/socket/pull/300, https://github.com/clue/reactphp-zenity/pull/63, https://github.com/clue/reactphp-csv/pull/33 and others.

After this is merged, I also want to create another pull request that fix the typos, improve the documentation. 👍

### CODE REVIEW

**Review by SimonFrings — CHANGES_REQUESTED** [review]
Looks good so far, I have two suggestions that are currently missing from this PR:

1. Can you also update the comment inside the `phpunit.xml.legacy`, similar to [my changes in ReactPHP CSV](https://github.com/clue/reactphp-csv/pull/33/files#diff-90e8ed4d404ab38c64ae8ee57d9df2b954891d9d57054048a77df0a385f3a612L3-R3)
2. We also use these PRs to improve the indentation inside the `composer.json`, similar to [my changes in ReactPHP CSV](https://github.com/clue/reactphp-csv/pull/33/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34L23-R30)

**Comment by SimonFrings** [comment]
It also seems like the tests for HHVM are currently failing, didn't notice while reviewing. Can you look into this?

**Comment by yadaiio** [comment]
> Looks good so far, I have two suggestions that are currently missing from this PR:
> 
>     1. Can you also update the comment inside the `phpunit.xml.legacy`, similar to [my changes in ReactPHP CSV](https://github.com/clue/reactphp-csv/pull/33/files#diff-90e8ed4d404ab38c64ae8ee57d9df2b954891d9d57054048a77df0a385f3a612L3-R3)
> 
>     2. We also use these PRs to improve the indentation inside the `composer.json`, similar to [my changes in ReactPHP CSV](https://github.com/clue/reactphp-csv/pull/33/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34L23-R30)

Done! Thank you @SimonFrings 👍

**Comment by yadaiio** [comment]
> It also seems like the tests for HHVM are currently failing, didn't notice while reviewing. Can you look into this?

It seems like we give an `array` in the React/Promise(any) function but HHVM expects an `React\Promise\Iterable` which doesn't exist. I decided to skip the tests for HHVM as the error message doesn't make much sense and HHVM is outdated anyway. Normally an `array` is the alias for `iterable`, see https://www.php.net/manual/en/language.types.iterable.php, so the thrown error message is invalid in my opinion.

Tests are running green now. 👍

errror

**Review by SimonFrings — APPROVED** [review]
Thanks for applying the suggested changes, PR looks good to me :+1:

File: tests/Multiple/ConnectionManagerConcurrentTest.php [hunk 1]
```diff
@@ -16,6 +16,10 @@ public function testEmptyListsThrows()
 
     public function testWillForwardToInnerConnector()
     {
+        if (defined('HHVM_VERSION')) {
+            $this->markTestSkipped('Not supported on HHVM with Promise v3');
+        }
```
Comment (clue) [inline tests/Multiple/ConnectionManagerConcurrentTest.php]: Looks like a valid option and HHVM definitely doesn't deserve any priority for this project. As an alternative, perhaps we should install Promise v2 for HHVM in our test setup instead? See https://github.com/reactphp/promise-timer/pull/67, https://github.com/reactphp/reactphp/pull/524 and others.

File: composer.json [hunk 2]
```diff
@@ -11,10 +11,14 @@
         }
     ],
     "autoload": {
-        "psr-4": { "ConnectionManager\\Extra\\": "src" }
+        "psr-4": {
+            "ConnectionManager\\Extra\\": "src"
```
Comment (clue) [inline composer.json]: If you're going to update this line, I suggest adding a trailing directory slash for consistency reasons:

```suggestion
            "ConnectionManager\\Extra\\": "src/"
```

I would also suggest moving the autoload/autoload-dev sections below the require/require-dev sections for consistency reasons. WDYT?

Comment (yadaiio) [reply composer.json]: Yess, seems fine. I changed it, thanks @clue 👍

File: composer.json [hunk 3]
```diff
@@ -24,6 +28,16 @@
         "react/socket": "^1.12"
     },
     "require-dev": {
-        "phpunit/phpunit": "^9.3 || ^5.7 || ^4.8"
+        "phpunit/phpunit": "^9.6 || ^5.7 || ^4.8.36"
+        },
```
Comment (clue) [inline composer.json]:
```suggestion
    },
```

Comment (yadaiio) [reply composer.json]: Thanks @clue, changed it. 👍

**Review by SimonFrings — APPROVED** [review]
Thanks for applying the changes, looks good to me :+1:

File: composer.json [hunk 4]
```diff
@@ -24,6 +28,16 @@
...
+    "autoload": {
+        "psr-4": {
+            "ConnectionManager\\Extra\\": "src/"
+        }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "ConnectionManager\\Tests\\Extra\\": "tests/"
+        }
     }
```
Comment (clue) [inline composer.json:32]: @yadaiio Duplicate keys, see above. Moving to bottom of file makes sense to me otherwise.

**Review by clue — CHANGES_REQUESTED** [review]

File: composer.json (hunk 4 above)
Comment (yadaiio) [reply composer.json:32]: Oops, I improved this and I was moving the keys to the bottom of the file. Thank you @clue 👍

**Comment by yadaiio** [comment]
As today is my last day working with you @clue, we agreed closing this pull request and revisit this later on.

If anyone is needing these changes earlier than expected, please consider sponsoring @clue via https://github.com/sponsors/clue. ❤️

**Pull request closed without merging**